docs/data-sources/dli_spark_templates.md

```diff
@@ -0,0 +1,143 @@
+---
+subcategory: "Data Lake Insight (DLI)"
+layout: "huaweicloud"
+page_title: "HuaweiCloud: huaweicloud_dli_spark_templates"
+description: |-
+  Use this data source to get the list of the DLI spark templates.
+---
+
+# huaweicloud_dli_spark_templates
+
+Use this data source to get the list of the DLI spark templates.
+
+## Example Usage
+
+```hcl
+variable "template_id" {}
+
+data "huaweicloud_dli_spark_templates" "test" {
+  template_id = var.template_id
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region in which to query the resource.
+  If omitted, the provider-level region will be used.
+
+* `template_id` - (Optional, String) Specifies the ID of the spark template to be queried.
+
+* `name` - (Optional, String) Specifies the name of the spark template to be queried.
+
+* `group` - (Optional, String) Specifies the group name to which the spark templates belong.
+
+## Attribute Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The data source ID.
+
+* `templates` - All templates that match the filter parameters.
+
+  The [templates](#templates_struct) structure is documented below.
+
+<a name="templates_struct"></a>
+The `templates` block supports:
+
+* `description` - The description of template.
+
+* `id` - The ID of template.
+
+* `name` - The name of template.
+
+* `body` - The body of template.
+
+  The [body](#templates_body_struct) structure is documented below.
+
+* `group` - The group of template.
+
+<a name="templates_body_struct"></a>
+The `body` block supports:
+
+* `app_parameters` - The input parameters of the main class, that is application parameters.
+
+* `name` - The Spark job name.
+
+* `obs_bucket` - The OBS bucket for storing the Spark jobs.
+
+* `auto_recovery` - The whether to enable the retry function.
+
+* `max_retry_times` - The maximum number of retries.
+
+* `jars` - The name of the resource package of type jar upload to the DLI resource management system.
+
+* `dependent_packages` - The list of package resource objects.
+
+  The [dependent_packages](#body_dependent_packages_struct) structure is documented below.
+
+* `configurations` - The configuration items of the DLI Spark.
+
+* `executor_memory` - The executor memory of the Spark application.
+
+* `queue_name` - The DLI queue name.
+
+* `main_class` - The Spark main class of the template.
+
+* `python_files` - The name of the resource package of type pyFile upload to the DLI resource management system.
```

Comment:
Please refer to the documentation and modify the fields with ambiguous descriptions.
https://support.huaweicloud.com/intl/en-us/api-dli/dli_02_0354.html

Reply:
done